### PR TITLE
Extract Location object tests from query-encoding/

### PR DESCRIPTION
**Pull request by annevk**
Helps with #4934.

### CODE REVIEW

**Comment by annevk** [comment]
Bugs:
* https://bugs.webkit.org/show_bug.cgi?id=185419
* https://bugs.chromium.org/p/chromium/issues/detail?id=840760

**Comment by annevk** [comment]
@hsivonen this relies on defaulting to windows-1252. Is that a reasonable thing to do in a test suite?

**Comment by hsivonen** [comment]
I think it's reasonable enough. Ideally, the instance of the test suite available to ad hoc running by users would be hosted on a non-com/org/net domain that's also not affiliated with an non-windows-1252 encoding. E.g. a `.io` (of to avoid `.io`, any of the new vanity TLDs) would work better than the current `.org`.

File: html/infrastructure/urls/resolving-urls/query-encoding/location.sub.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+<meta name=variant content="?encoding=x-cp1251">
+<meta name=variant content="?encoding=utf8">
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<div id=log></div>
+<script>
+function expected(encoding) {
+  return "?" + {
+    "UTF-8": "%C3%BF",
+    "windows-1251": "&%23255;",
```
Comment (annevk) [inline html/infrastructure/urls/resolving-urls/query-encoding/location.sub.html]: Either change this to align with https://github.com/whatwg/url/pull/386 before landing or rebase #10915 and change it there.

**Comment by foolip** [comment]
Interesting, I came across this part of the test suite in https://github.com/web-platform-tests/wpt.fyi/issues/132 by a completely different way of looking for problems. @zcorpan FYI.

**Review by foolip — APPROVED** [review]
LGTM assuming the tests still work :)

**Comment by annevk** [comment]
@zcorpan could you quickly skim this as well?

**Comment by zcorpan** [comment]
Looks good to me.

**Pull request merged**